enigma-swing/src/main/java/cuchaz/enigma/gui/panels/right/RightPanel.java

```diff
@@ -0,0 +1,75 @@
+package cuchaz.enigma.gui.panels.right;
+
+import cuchaz.enigma.gui.Gui;
+import cuchaz.enigma.utils.I18n;
+
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JToggleButton;
+import java.awt.BorderLayout;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public abstract class RightPanel extends JPanel {
```

Comment:
Found a bug. Seems like the width for the panels reset to 0 when changing between panels under some conditions. Reproduction steps:
1. Launch enigma
2. Open a right panel if one isn't there already
3. Click on the button for the selected panel to close it
4. Open a different panel
5. Click on the button for the previous panel (same on as step 4)
6. A panel won't appear, until you try using the 3 dots to resize it